project.mml

```diff
@@ -475,12 +475,12 @@ Layer:
             ELSE NULL
           END AS maxspeed_kmh,
           CASE
-            WHEN (
-              tags->'psv' != 'no' OR tags->'motorcar' != 'no' OR tags->'bus' != 'no'
-              OR tags->'motor_vehicle' != 'no'
-              OR tags->'vehicle' != 'no'
-              ) THEN 'yes'
-            WHEN tags->'psv' IS NULL AND tags->'motorcar' IS NULL AND tags->'bus' IS NULL AND tags->'motor_vehicle' IS NULL AND tags->'vehicle' IS NULL AND (access != 'no' OR access IS NULL) THEN 'yes'
+            WHEN COALESCE(motorcar, tags->'motor_vehicle', tags->'vehicle', access, 'yes') != 'no' THEN 'yes'
+            -- goods and hgv are not technically children of motorcar, but the motorcar restriction is often intended to cover cars and all larger vehicles
+            WHEN COALESCE(tags->'goods', motorcar, tags->'motor_vehicle', tags->'vehicle', access, 'yes') != 'no' THEN 'yes'
+            WHEN COALESCE(tags->'hgv', motorcar, tags->'motor_vehicle', tags->'vehicle', access, 'yes') != 'no' THEN 'yes'
+            WHEN COALESCE(tags->'motorcycle', tags->'motor_vehicle', tags->'vehicle', access, 'yes') != 'no' THEN 'yes'
+            -- TODO: style psv-only roads slightly differently
```

Comment:
@Florimondable what is your opinion on this? PSV-only roads are basically `highway=busway` (this tag does not exist). Maybe these could be rendered as no motor vehicle?

E.g.: https://www.openstreetmap.org/way/202871519#map=18/48.72352/2.24382

Reply:
PSV-only roads have motor vehicle on it, so no-motor color can't be applied ;)

Your example is a no bicycle road btw

Reply:
Even if nomotor isn't the correct style, it could be worth defining a separate style for shared-with-PSV roads, similar to how cycleway=share_busway gets a style similar to, but inferior to, cycleway=lane. I've left a TODO, while making the current behaviour that these roads are styled as if cars were allowed on them.